File: 프로그래머스/unrated/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.py
def solution(t, p):
    answer = 0

    numbers = list(map(int, list(t)))
    compares = list(map(int, list(p)))
    len_compares = len(compares)

    for i in range(len(t) - len_compares + 1):
        l = 0
        while l < len_compares:
            if numbers[i + l] > compares[l]:
                break
            elif numbers[i + l] == compares[l]:
                l += 1
            else:  # numbers[i + l] < compares[l]
                l = len_compares  # go to the next sequence
                
        if l == len_compares:
            answer += 1

    return answer

File: 프로그래머스/unrated/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.js
function solution(t, p) {
    var answer = 0;
    
    const numbers = t.split('').map(v=>+v)
    const compares = p.split('').map(v=>+v)
    const lenCompares = compares.length

    for (let i=0; i<=t.length-lenCompares; i++) {
        const n = numbers[i]
        
        let l = 0
        while (l<lenCompares) {
            if (numbers[i+l] < compares[l]) {
                l = lenCompares
                break;
            } else if (numbers[i+l] === compares[l]) {
                l ++
            } else {
                break;
            }
        }
            
        if (l === lenCompares) answer ++
    }
    
    return answer;
}